src/test/java/seedu/exercise/logic/parser/AddCommandParserTest.java

```diff
@@ -0,0 +1,174 @@
+import static seedu.exercise.logic.commands.CommandTestUtil.VALID_NAME_BASKETBALL;
+import static seedu.exercise.logic.commands.CommandTestUtil.VALID_QUANTITY_BASKETBALL;
+import static seedu.exercise.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static seedu.exercise.logic.parser.CommandParserTestUtil.assertParseSuccess;
+import static seedu.exercise.testutil.TypicalExercises.AEROBICS;
+import static seedu.exercise.testutil.TypicalExercises.BASKETBALL;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.exercise.logic.commands.AddCommand;
+import seedu.exercise.model.exercise.Calories;
+import seedu.exercise.model.exercise.Date;
+import seedu.exercise.model.exercise.Exercise;
+import seedu.exercise.model.exercise.Name;
+import seedu.exercise.model.exercise.Quantity;
+import seedu.exercise.model.exercise.Unit;
+import seedu.exercise.model.tag.Muscle;
+import seedu.exercise.testutil.ExerciseBuilder;
+
+public class AddCommandParserTest {
+    private AddCommandParser parser = new AddCommandParser();
+
+    @Test
+    public void parse_allFieldsPresent_success() {
+        Exercise expectedExercise = new ExerciseBuilder(BASKETBALL).withMuscles(VALID_MUSCLE_BASKETBALL).build();
+
+        // whitespace only preamble
+        assertParseSuccess(parser, PREAMBLE_WHITESPACE + NAME_DESC_BASKETBALL
+            + DATE_DESC_BASKETBALL + CALORIES_DESC_BASKETBALL
+            + QUANTITY_DESC_BASKETBALL + UNIT_DESC_BASKETBALL + MUSCLE_DESC_BASKETBALL,
+            new AddCommand(expectedExercise));
+
+        // multiple names - last name accepted
+        assertParseSuccess(parser, NAME_DESC_AEROBICS + NAME_DESC_BASKETBALL
+            + DATE_DESC_BASKETBALL + CALORIES_DESC_BASKETBALL
+            + QUANTITY_DESC_BASKETBALL + UNIT_DESC_BASKETBALL + MUSCLE_DESC_BASKETBALL,
+            new AddCommand(expectedExercise));
+
+        // multiple dates - last date accepted
+        assertParseSuccess(parser, NAME_DESC_BASKETBALL + DATE_DESC_AEROBICS
+            + DATE_DESC_BASKETBALL + CALORIES_DESC_BASKETBALL
+            + QUANTITY_DESC_BASKETBALL + UNIT_DESC_BASKETBALL + MUSCLE_DESC_BASKETBALL,
+            new AddCommand(expectedExercise));
+
+        // multiple calories - last calories accepted
+        assertParseSuccess(parser, NAME_DESC_BASKETBALL + DATE_DESC_BASKETBALL
+            + CALORIES_DESC_AEROBICS + CALORIES_DESC_BASKETBALL
+            + QUANTITY_DESC_BASKETBALL + UNIT_DESC_BASKETBALL + MUSCLE_DESC_BASKETBALL,
+            new AddCommand(expectedExercise));
+
+        // multiple quantities - last quantity accepted
+        assertParseSuccess(parser, NAME_DESC_BASKETBALL + DATE_DESC_BASKETBALL
+            + CALORIES_DESC_BASKETBALL + QUANTITY_DESC_AEROBICS
+            + QUANTITY_DESC_BASKETBALL + UNIT_DESC_BASKETBALL + MUSCLE_DESC_BASKETBALL,
+            new AddCommand(expectedExercise));
+
+        // multiple quantities - last unit accepted
```

Comment:
The comment here should be `multiple units - list unit accepted`

Reply:
done